apps/minaintyg/src/pages/Certificates/components/CertificateListFilter/CertificateListFilter.test.tsx

```diff
@@ -0,0 +1,87 @@
+import { fakeCertificateId, faker, fakerFromSchema } from '@frontend/fake'
+import { render, screen, waitFor, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { rest } from 'msw'
+import { Provider } from 'react-redux'
+import { vi } from 'vitest'
+import { server } from '../../../../mocks/server'
+import { CertificateStatusEnum } from '../../../../schema/certificateList.schema'
+import { certificateFilterOptionsSchema } from '../../../../schema/certificateListFilter.schema'
+import { store } from '../../../../store/store'
+import { CertificateListFilter } from './CertificateListFilter'
+
+const options = {
+  statuses: CertificateStatusEnum.options,
+  certificateTypes: faker.helpers.uniqueArray(fakeCertificateId, 4),
+  units: faker.helpers.uniqueArray(faker.company.name, 4),
+  years: ['2023', '2022', '2021', '2020'],
+}
+
+beforeEach(() => {
+  server.use(
+    rest.get('/api/filter-certificate', (_, res, ctx) =>
+      res(ctx.status(200), ctx.json(fakerFromSchema(certificateFilterOptionsSchema)(options)))
+    )
+  )
+})
+
+describe('Options from API', () => {
+  beforeEach(() => {
+    render(
+      <Provider store={store}>
+        <CertificateListFilter onSubmit={vi.fn()} />
+      </Provider>
+    )
+  })
+
+  it.each(['Status', 'Mottagning', 'Intygstyp', 'År'])('Should have corret %s option', async (name) => {
+    expect(
+      (within(screen.getByLabelText(name)).getByRole('option', { name: `Välj ${name.toLowerCase()}` }) as HTMLOptionElement).selected
+    ).toBe(true)
+    waitFor(() => {
+      expect(within(screen.getByLabelText(name)).getAllByRole('option').length).toBe(CertificateStatusEnum.options.length + 1)
+    })
+  })
+})
+
+it('Should call on submit when "Filtrera" is pressed', async () => {
+  const onSubmit = vi.fn()
+  render(
+    <Provider store={store}>
+      <CertificateListFilter onSubmit={onSubmit} />
+    </Provider>
+  )
+
+  await userEvent.click(screen.getByLabelText('Filtrera'))
+
+  expect(onSubmit).toHaveBeenCalledTimes(1)
+})
+
+it('Should reset filters when "Återställ filter" is pressed', async () => {
```

Comment:
This test is long and a bit hard to understand. Maybe it can be made easier to understand if some parts were extracted to functions and the test was split up into several tests.

Reply:
Rewrote it using `.each` to separate the filters, hopefully more readable now.